src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -347,8 +382,26 @@ internal override uint GenerateSspiClientContext(byte[] receivedBuff, uint recei
 
         internal override uint WaitForSSLHandShakeToComplete(out int protocolVersion)
         {
-            protocolVersion = Handle.ProtocolVersion;
+            protocolVersion = GetSessionSNIHandleHandleOrThrow().ProtocolVersion;
             return 0;
         }
+
+        private SNIHandle GetSessionSNIHandleHandleOrThrow()
+        {
+            SNIHandle? sessionHandle = _sessionHandle;
+            if (sessionHandle is null)
+            {
+                ThrowClosedConnection();
+            }
+            return sessionHandle!;
+        }
+
+#if NETCOREAPP
+        [DoesNotReturn]
+#endif 
```

Comment:
We could remove this ifdef if we added a nuget reference which implements the reference, there is one available. I'm not sure if it is a good idea to do this because the nullable attributes are a compile time feature so passing the dependency on to users doesn't seem useful to me.